### ISSUE TITLE
No way to set broker.SubscriberOptions

### ISSUE THREAD

**Issue by alethenorio** [micro/go-micro#398]
Upon looking at creating an example which reads messages from a broker, performs work on it (Such as writing to a database) and exposes grpc methods, I could not find a way to configure all possible broker options.

The idea was to do the following

```go
package main

import (
	"flag"
	"log"

	"github.com/micro/go-micro/server"

	micro "github.com/micro/go-micro"
    "github.com/micro/go-plugins/broker/rabbitmq"
)

func main() {

    rmq := rabbitmq.NewBroker()

	// Create a new service. Optionally include some options here.
	service := micro.NewService(
		micro.Name("greeter"),
		micro.Broker(rmq),
	)

	// Init will parse the command line flags.
	service.Init()

	// Register handler
	tags.RegisterGreeterHandler(service.Server(), new(Greeter))

	micro.RegisterSubscriber("#", service.Server(), ProcessGreets, server.SubscriberQueue("greets"))

	// Run the server
	if err := service.Run(); err != nil {
		log.Fatal(err)
	}
}

func ProcessGreets(ctx context.Context, event *greeter.Greet) error {
	fmt.Printf("Received tag %v\n", event)
	return nil
}

```

It seems that the `broker.SubscribeOptions` struct allows for configuring certain properties of the broker such as AutoAck (There are even helper functions such as `broker.DisableAutoAck()`) however when trying to create a subscriber there seems to be no way to configure the subscriber `broker.SubscriberOptions`. As a consequence it does not seem to be possible to configure autoack at the moment and it is always true (Which might not be desired as it is clearly shown in the `rabbitmq` plugin which handles such scenario, should it be possible to configure it). The `micro.RegisterSubscriber` method allows for passing `server.SubscriberOption` which does not expose all possible options.

Should that option not be exposed for the service to configure it?

**Comment by vtolstov** [comment]
this is already discussed on slack yesterday , currently this is not possible
@asim says that he need time to rethink this.

**Comment by alethenorio** [comment]
That's too bad. At least the plugins were written with the option for
configuration in mind and hopefully the framework can catch to that soon.

Thanks for the heads up

**Comment by vtolstov** [comment]
i'm finish nats streaming broker, and when i push it to go-plugins i can provide example what need to be changed

**Comment by vtolstov** [comment]
broker need to have SubscribeContext broker.SubscribeOption (https://github.com/micro/go-plugins/pull/269/files#diff-39f6b80afedae06744d49a3bf01e1e71R28)
and in Subscribe function you need to use passed context (https://github.com/micro/go-plugins/pull/269/files#diff-a87c433880940bfa4fc9b5ba58d4d777R209)

**Comment by vtolstov** [comment]
may be i'm or somebody else fix other brokers

**Comment by asim** [comment]
This change was merged to include subscribe options https://github.com/micro/go-micro/commit/67a738b504aa2c289f3f3a353322338e073a38dd

If you want to set broker options please preconfigure the broker before passing it in.

**Comment by yexm** [comment]
> This change was merged to include subscribe options [67a738b](https://github.com/micro/go-micro/commit/67a738b504aa2c289f3f3a353322338e073a38dd)
> 
> If you want to set broker options please preconfigure the broker before passing it in.

i found rabbitmq can not set the durableQueue of broker options. it only can set the name of queue and AutoAck.
